### ISSUE TITLE
Wrong Outline of no level 0 is present, default is 1

### ISSUE THREAD

**Issue by rspilker** [DavBfr/dart_pdf#892]
**Describe the bug**
In my document I add a `Header(level: 1, text: 'Main')` followed by `Header(level: 2, text: 'Sub 1')` and  `Header(level: 2, text: 'Sub 2')`. 

In the generated pdf, I now get three levels, and _Main_ only has one direct child, _Sub 1_, and _Sub 1_ becomes the parent of _Sub 2_. Adding `Header(level: 2, text: 'Sub 3')` would be placed in the same level as _Sub 2_, as a child of _Sub 1_.

**To Reproduce**
Code snippet to reproduce the behavior:
```dart
import 'dart:io';
import 'package:pdf/pdf.dart';
import 'package:pdf/widgets.dart' as pw;

void main() async {
  final pdf = pw.Document();

  pdf.addPage(pw.Page(build: (pw.Context context) {
    return pw.Column(
      children: [
        _header('Top level 1', 1),
        _header('Sub level 1.1', 2),
        _header('Sub level 1.2', 2),
        _header('Sub level 1.3', 2),
        _header('Sub level 1.4', 2),
        _header('Top level 2', 1),
        _header('Sub level 2.1', 2),
        _header('Sub level 2.2', 2),
        _header('Sub level 2.3', 2),
      ],
    );
  }));
  final file = File('test-outline.pdf');
  await file.writeAsBytes(await pdf.save());
}

pw.Widget _header(String text, int level) => pw.Header(
      text: text,
      level: level,
    );
```

**Expected behavior**
Instead of generating

- Top level 1
  - Sub level 1.1
    - Sub level 1.2
    - Sub level 1.3
    - Sub level 1.4
- Top level 2
  - Sub level 2.1
    - Sub level 2.2
    - Sub level 2.3

I expected to get

- Top level 1
  - Sub level 1.1
  - Sub level 1.2
  - Sub level 1.3
  - Sub level 1.4
- Top level 2
  - Sub level 2.1
  - Sub level 2.2
  - Sub level 2.3

**Screenshots**
![outline-level-bug](https://user-images.githubusercontent.com/93318/147712331-5c408fb5-5322-46bc-abeb-f18c788579c1.png)

**Generated file**
[test-outline.pdf](https://github.com/DavBfr/dart_pdf/files/7791071/test-outline.pdf)

**Relevant section from pdf**
```
7 0 obj
<</Title(Top level 1)/Dest(999710566)/Parent 8 0 R/Count -4/Next 13 0 R/First 9 0 R/Last 9 0 R>>
endobj
8 0 obj
<</Count 2/First 7 0 R/Last 13 0 R>>
endobj
9 0 obj
<</Title(Sub level 1.1)/Dest(656008326)/Parent 7 0 R/Count -3/First 10 0 R/Last 12 0 R>>
endobj
10 0 obj
<</Title(Sub level 1.2)/Dest(213528612)/Parent 9 0 R/Next 11 0 R>>
endobj
11 0 obj
<</Title(Sub level 1.3)/Dest(1053791311)/Parent 9 0 R/Prev 10 0 R/Next 12 0 R>>
endobj
12 0 obj
<</Title(Sub level 1.4)/Dest(823523547)/Parent 9 0 R/Prev 11 0 R>>
endobj
13 0 obj
<</Title(Top level 2)/Dest(205226166)/Parent 8 0 R/Count -3/Prev 7 0 R/First 14 0 R/Last 14 0 R>>
endobj
14 0 obj
<</Title(Sub level 2.1)/Dest(752855642)/Parent 13 0 R/Count -2/First 15 0 R/Last 16 0 R>>
endobj
15 0 obj
<</Title(Sub level 2.2)/Dest(1059213023)/Parent 14 0 R/Next 16 0 R>>
endobj
16 0 obj
<</Title(Sub level 2.3)/Dest(108879254)/Parent 14 0 R/Prev 15 0 R>>
endobj
```

**Flutter Doctor**
- No flutter, plain dart on the command line

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [ ] Browser
 - [x] Windows
 - [ ] Linux

**Smartphone (please complete the following information):**
- No smartphone

**Additional information**
I'm willing to submit a pull request for fixing the code, but would need some help to also adjust the automated tests, like golden files.

The moment I added a level 0 header before the level 1, or just reduced all levels by one, the problem disappeared.

That does not mean that is is confusing, and there's still a bug somewhere.

**Comment by DavBfr** [comment]
You can write some tests. I'll update the golden files.

**Comment by DavBfr** [comment]
Merged to pdf-3.7.0
